Caleydo/tdp_publicdb
# Integrate alembic and add hg38 DB schema as initial revision

This PR depeneds on datavisyn/tdp_core#292

Closes Caleydo/tdp_publicdb#64

### Summary
* Integrated alembic according to the instructions in https://wiki.datavisyn.io/tdp/extension-points/db-migration.
* Exported the _hg_38_ schema from the _postgres 12_ database and imported it  in the upgrade function  to be executed as the initial revision.
* Tested running `db-migration:upgrade:head:tdp_publicdb` both in a _postgres 10-alpine_ container and and  _postgres 12.1_ container. In both cases the schema gets imported and the _tdp_publicdb_alembic_version_ table with the current revision ID gets created in the public schema.


The following script to create the _tdp_publicdb_alembic_version_ table needs to be executed in the hg_38 postgres 12.1 database so the alembic upgrade command does not try to upgrade it.

```sql
CREATE TABLE public.tdp_publicdb_alembic_version
(
    version_num character varying(32) COLLATE pg_catalog."default" NOT NULL,
    CONSTRAINT tdp_publicdb_alembic_version_pkc PRIMARY KEY (version_num)
)
WITH (
    OIDS = FALSE
)
TABLESPACE pg_default;

ALTER TABLE public.tdp_publicdb_alembic_version
    OWNER to publicdb;

INSERT INTO public.tdp_publicdb_alembic_version(version_num)
	VALUES('b477004c3673');
```

* Already tested the above script in a 12 and 10 postgres container.

#### To test

1. ~Switch  tdp_core branch to the `mp/291_db_migration` (necessary since the alembic integration is not yet in the develop branch).~ --> merged into develop
2. Create a `postgres12.1` container and a matching `pgadmin` container in your `docker-compose.yml` file.
3. Open `pgadmin` on the browser and add _ordino_ and _postgres_ users/roles.
4. Execute alembic command `db-migration:upgrade:head:tdp_publicdb`.
5. Check if the schema looks okay and also that  the table `tdp_publicdb_alembic_version` has been created.
